actors/user/src/test/java/org/sunbird/user/UserRoleActorTest.java

```diff
@@ -0,0 +1,252 @@
+    when(BaseMWService.getRemoteRouter(Mockito.anyString())).thenReturn(actorSelection);
+    when(actorSelection.resolveOneCS(Duration.create(Mockito.anyLong(), "seconds")))
+        .thenReturn(completionStage);
+
+    ActorRef actorRef = Mockito.mock(ActorRef.class);
+    when(RequestRouter.getActor(Mockito.anyString())).thenReturn(actorRef);
+
+    searchDTO = Mockito.mock(SearchDTO.class);
+    when(Util.createSearchDto(Mockito.anyMap())).thenReturn(searchDTO);
+
+    cassandraOperation = mock(CassandraOperationImpl.class);
+    when(ServiceFactory.getInstance()).thenReturn(cassandraOperation);
+  }
+
+  @Test
+  public void testGetUserRoleSuccess() {
+
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+    when(cassandraOperation.getAllRecords(Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(getCassandraResponse());
+    Request reqObj = new Request();
+    reqObj.setOperation(ActorOperations.GET_ROLES.getValue());
+    subject.tell(reqObj, probe.getRef());
+    Response res = probe.expectMsgClass(duration("10 second"), Response.class);
+    Assert.assertTrue(null != res && res.getResponseCode() == ResponseCode.OK);
+  }
+
+  @Test
+  public void testAssignRolesSuccessWithValidOrgId() {
+
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+    mockGetOrgResponse(true);
+    Map<String, Object> map = Mockito.mock(Map.class);
+    when(interServiceCommunication.getResponse(Mockito.anyObject(), Mockito.anyObject()))
+        .thenReturn(response);
+    when(response.get(Mockito.anyString())).thenReturn(map);
+    subject.tell(getRequestObj(true), probe.getRef());
+    Response res = probe.expectMsgClass(duration("10 second"), Response.class);
+    Assert.assertTrue(null != res && res.getResponseCode() == ResponseCode.OK);
+  }
+
+  @Test
+  public void testAssignRolesSuccessWithoutOrgId() {
+
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+    Map<String, Object> map = Mockito.mock(Map.class);
+    when(interServiceCommunication.getResponse(Mockito.anyObject(), Mockito.anyObject()))
+        .thenReturn(response);
+    when(response.get(Mockito.anyString())).thenReturn(map);
+    mockGetOrgResponse(true);
+    decryptionService = Mockito.mock(DecryptionService.class);
+    when(decryptionService.decryptData(Mockito.anyMap())).thenReturn(getOrganisationsMap());
+    subject.tell(getRequestObj(false), probe.getRef());
+    Response res = probe.expectMsgClass(duration("10 second"), Response.class);
+    Assert.assertTrue(null != res && res.getResponseCode() == ResponseCode.OK);
+  }
+
+  @Test
+  public void testAssignRolesFailure() {
+
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+    mockGetOrgResponse(false);
+    subject.tell(getRequestObj(true), probe.getRef());
+    ProjectCommonException res =
```

Comment:
in each test case 4 lines are repeated - atleast 
there is a pattern
- initialize probe
- initilize subject
- subject.tell
- probe.expectMsgClass